### PR TITLE
[Merged by Bors] - feat(Mathlib/Order/Basic): port most of data/order/basic.lean

### PR DESCRIPTION
**Pull request by dwrensha**
Ports most of [data/order/basic.lean](https://github.com/leanprover-community/mathlib/blob/master/src/order/basic.lean) from mathlib3, leaving a few TODOs.

This file lives near the base of the file-import dependency graph; having it ported should enable more experimentation with higher-level parts of the algebraic hierarchy.

Some challenges that came up:
 - Lean 4 defines `max` and `min` in the prelude, while Lean 3 (as of https://github.com/leanprover-community/lean/pull/609) defines them differently as part of `linear_order`. We will need to figure out how to resolve this conflict.
 - In Lean 3, the lexicographic order on products uses a wrapper type `lex α β`. In Lean 4, the prelude instance `instLTProd` defines the lexicographic order directly on `α × β`. In this PR, I needed to verbosely disambiguate which instance to use.

### CODE REVIEW

File: scripts/nolints.json [hunk 1]
```diff
@@ -86,11 +86,16 @@
  ["docBlame", "termSudoSet_option___In_"],
  ["docBlame", "«term_\\_»"],
  ["docBlame", "«term_⁻¹»"],
+ ["docBlame", "«term_⁻¹'o_»"],
```
Comment (gebner) [inline scripts/nolints.json:89]: Please document the syntax instead of silencing the lint.

File: Mathlib/Order/Basic.lean [hunk 2]
```diff
@@ -0,0 +1,731 @@
...
+protected lemma ge (h : x = y) : y ≤ x := h.symm.le
+
+lemma not_lt (h : x = y) : ¬ x < y := λ h' => h'.ne h
+lemma not_gt (h : x = y) : ¬ y < x := h.symm.not_lt
+
+end Eq
+
+namespace LE.le
+
+@[nolint ge_or_gt] -- see Note [nolint_ge]
```
Comment (gebner) [inline Mathlib/Order/Basic.lean:156]:
```suggestion
```

The lint no longer exists.

File: scripts/nolints.json (hunk 1 above)
Comment (dwrensha) [reply scripts/nolints.json:89]: When I try to do that I get a linter error:
```
/- The `docBlame` linter reports:
DEFINITIONS ARE MISSING DOCUMENTATION STRINGS: -/
-- Mathlib.Order.Basic
#check term_⁻¹'o_ /- definition missing documentation string -/
```

Comment (digama0) [reply scripts/nolints.json:89]: yes, because the definition is missing a doc string. You should fix it by adding one instead of silencing the lint.

Comment (dwrensha) [reply scripts/nolints.json:89]: Aha, my problem was that I wasn't re-running `lake build`, I was just running the linter.

Comment (digama0) [reply scripts/nolints.json:89]: Note: in recent versions of lean 4 core the `@[inheritDoc]` attribute exists to copy the doc string onto `infix` and `notation` declarations, and we will want to roll it out in mathlib4 along with the bump commit, which should fix this issue.

**Comment by gebner** [comment]
bors merge

**Pull request closed without merging**